src/NLog/Internal/FileAppenders/CountingSingleProcessFileAppender.cs

```diff
@@ -98,7 +111,14 @@ public override void Flush()
                 return;
             }
 
-            this.file.Flush();
+            try
+            {
+                this.file.Flush();
+            }
+            catch (Exception ex)
+            {
+                throw new System.IO.IOException("FileStream Flush Failed: " + FileName, ex);
```

Comment:
I'm not sure if this is how I would implement it.

Isn't better to just rethrow the ex, and maybe add some context to the exception? e.g. 

```c#
catch (Exception ex)
{
    ex.data["info"] = "FileStream Flush Failed: " + FileName;
    throw;
} 
```